globals.set action does not respect variable type

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:

- Provide as many details as possible. Simply saying "X gives bug" or "X gives error" is not enough!
- Paste logs, configuration sample and code into the backticks (```).
- Read through the template carefully and fill out all missing details.
- Please also search for similar issues in this issue tracker first and read through the ESPHome FAQ.

DO NOT DELETE ANY TEXT from this template! Otherwise the issue may be closed without a comment.
-->

**Operating environment/Installation (Hass.io/Docker/pip/etc.):**
<!--
Please provide details about your environment below this line. -->
pip (esphome-1.13.5)

**ESP (ESP32/ESP8266, Board/Sonoff):**
<!--
Please provide details about which ESP you're using below.
-->
ESP8266

**Affected component:**
<!--
Please add the link to the documentation at https://esphome.io/index.html of the component in question.
-->
globals

**Description of problem:**
Using the globals.set action with an int type global results in a compile error

**Problem-relevant YAML-configuration entries:**
```yaml
esphome:
  name: test_globalsset_bug
  platform: ESP8266
  board: nodemcuv2

logger:

globals:
  - id: test1
    type: int

script:
  - id: test_script
    then:
      - globals.set:
          id: test1
          value: '1'
```

**Traceback (if applicable):**
<!--
Please copy the traceback here if compilation is failing. If possible, also connect to the ESP over USB and copy its logs into the backticks.
-->
```
$ esphome bugreport.yaml compile
INFO Reading configuration...
INFO Generating C++ source...
INFO Compiling app...
INFO Running:  platformio run -d test_globalsset_bug
Processing test_globalsset_bug (framework: arduino; platform: espressif8266@1.8.0; board: nodemcuv2)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/nodemcuv2.html
PLATFORM: Espressif 8266 > NodeMCU 1.0 (ESP-12E Module)
HARDWARE: ESP8266 80MHz 80KB RAM (4MB Flash)
Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF MODES: FINDER(chain) COMPATIBILITY(soft)
Collected 26 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <ESP8266WiFi> 1.0
|-- <ESP8266mDNS>
|   |-- <ESP8266WiFi> 1.0
Compiling .pioenvs/test_globalsset_bug/src/main.cpp.o
Compiling .pioenvs/test_globalsset_bug/lib9a9/ESP8266WiFi/ESP8266WiFiAP.cpp.o
Compiling .pioenvs/test_globalsset_bug/lib9a9/ESP8266WiFi/ESP8266WiFiGeneric.cpp.o
Compiling .pioenvs/test_globalsset_bug/lib9a9/ESP8266WiFi/ESP8266WiFiMulti.cpp.o
In file included from src/esphome/core/automation.h:5:0,
from src/esphome/components/globals/globals_component.h:4,
from src/esphome.h:2,
from src/main.cpp:3:
src/esphome/core/helpers.h: In instantiation of 'esphome::TemplatableValue<T, X>::TemplatableValue(F) [with F = const char*; typename std::enable_if<(! esphome::is_callable<F, X ...>::value), int>::type <anonymous> = 0; T = int; X = {}]':
src/esphome/components/globals/globals_component.h:59:3:   required from 'void esphome::globals::GlobalVarSetAction<C, Ts>::set_value(V) [with V = const char*; C = esphome::globals::GlobalsComponent<int>; Ts = {}]'
src/main.cpp:56:44:   required from here
src/esphome/core/helpers.h:202:57: error: invalid conversion from 'const char*' to 'int' [-fpermissive]
TemplatableValue(F value) : type_(VALUE), value_(value) {}
^
*** [.pioenvs/test_globalsset_bug/src/main.cpp.o] Error 1
================================================================================================================= [ERROR] Took 2.26 seconds =================================================================================================================
```

**Additional information and things you've tried:**
Inclusion of initial_value doesn't seem to make any difference.



<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

--- a/esphome/components/globals/__init__.py
+++ b/esphome/components/globals/__init__.py
@@ -50,7 +50,7 @@ def globals_set_to_code(config, action_id, template_arg, args):
     full_id, paren = yield cg.get_variable_with_full_id(config[CONF_ID])
     template_arg = cg.TemplateArguments(full_id.type, *template_arg)
     var = cg.new_Pvariable(action_id, template_arg, paren)
-    templ = yield cg.templatable(config[CONF_VALUE], args,
+    templ = yield cg.templatable(config[CONF_VALUE], args, None,
                                  to_exp=cg.RawExpression)
     cg.add(var.set_value(templ))
     yield var

--- a/esphome/components/globals/__init__.py
+++ b/esphome/components/globals/__init__.py
@@ -50,6 +50,7 @@ def globals_set_to_code(config, action_id, template_arg, args):
     full_id, paren = yield cg.get_variable_with_full_id(config[CONF_ID])
     template_arg = cg.TemplateArguments(full_id.type, *template_arg)
     var = cg.new_Pvariable(action_id, template_arg, paren)
-    templ = yield cg.templatable(config[CONF_VALUE], args, None)
+    templ = yield cg.templatable(config[CONF_VALUE], args,
+                                 to_exp=cg.RawExpression)
     cg.add(var.set_value(templ))
     yield var

--- a/tests/test1.yaml
+++ b/tests/test1.yaml
@@ -1277,6 +1277,9 @@ text_sensor:
       id: template_text
       state: |-
         return "Hello World2";
+  - globals.set:
+      id: glob_int
+      value: '0'
 - platform: template
   name: Template Text Sensor
   id: template_text

